### PR TITLE
Add support for Wechat Pay

### PR DESCRIPTION
**Pull request by richardm-stripe**
## Changelog
* Add support for `WechatPay` on `ChargePaymentMethodDetails`, `CheckoutSessionPaymentMethodOptionsParams`, `PaymentIntentPaymentMethodDataParams`, `PaymentIntentPaymentMethodOptionsParams`, `PaymentIntentPaymentMethodOptions`, `PaymentMethodParams`, and `PaymentMethod`
* Add support for new value `wechat_pay` on enums `InvoicePaymentSettingsPaymentMethodType` and `PaymentMethodType`
* Add support for `WechatPayDisplayQRCode`, `WechatPayRedirectToAndroidApp`, and `WechatPayRedirectToIOSApp` on `PaymentIntentNextAction`

### CODE REVIEW

File: charge.go [hunk 1]
```diff
@@ -551,6 +551,10 @@ type ChargePaymentMethodDetailsStripeAccount struct {
 type ChargePaymentMethodDetailsWechat struct {
 }
 
+// ChargePaymentMethodDetailsWechatPay represents details about the WechatPay PaymentMethod.
+type ChargePaymentMethodDetailsWechatPay struct {
+}
```
Comment (remi-stripe) [inline charge.go:556]: missing `Fingerprint` and `TransactionID`

File: paymentintent.go [hunk 2]
```diff
@@ -110,6 +110,14 @@ const (
 	PaymentIntentPaymentMethodOptionsCardRequestThreeDSecureAutomatic PaymentIntentPaymentMethodOptionsCardRequestThreeDSecure = "automatic"
 )
 
+type PaymentIntentPaymentMethodOptionsWechatPayClient string
+
+const (
+	PaymentIntentPaymentMethodOptionsWechatPayClientAndroid PaymentIntentPaymentMethodOptionsWechatPayClient = "android"
+	PaymentIntentPaymentMethodOptionsWechatPayClientIos     PaymentIntentPaymentMethodOptionsWechatPayClient = "ios"
```
Comment (remi-stripe) [inline paymentintent.go:117]: should it be IOS?

File: paymentintent.go [hunk 3]
```diff
@@ -395,15 +412,43 @@ type PaymentIntentNextActionVerifyWithMicrodeposits struct {
...
+	AlipayHandleRedirect          *PaymentIntentNextActionAlipayHandleRedirect          `json:"alipay_handle_redirect"`
+	BoletoDisplayDetails          *PaymentIntentNextActionBoletoDisplayDetails          `json:"boleto_display_details"`
+	OXXODisplayDetails            *PaymentIntentNextActionOXXODisplayDetails            `json:"oxxo_display_details"`
+	RedirectToURL                 *PaymentIntentNextActionRedirectToURL                 `json:"redirect_to_url"`
+	Type                          PaymentIntentNextActionType                           `json:"type"`
+	UseStripeSDK                  *PaymentIntentNextActionUseStripeSDK                  `json:"use_stripe_sdk"`
+	VerifyWithMicrodeposits       *PaymentIntentNextActionVerifyWithMicrodeposits       `json:"verify_with_microdeposits"`
+	WechatPayDisplayQrCode        *PaymentIntentNextActionWechatPayDisplayQRCode        `json:"wechat_pay_display_qr_code"`
+	WechatPayRedirectToAndroidApp *PaymentIntentNextActionWechatPayRedirectToAndroidApp `json:"wechat_pay_redirect_to_android_app"`
+	WechatPayRedirectToIosApp     *PaymentIntentNextActionWechatPayRedirectToIOSApp     `json:"wechat_pay_redirect_to_ios_app"`
```
Comment (remi-stripe) [inline paymentintent.go:451]: same IOS here

**Review by remi-stripe — COMMENTED** [review]
flagging some issues

File: charge.go (hunk 1 above)
Comment (richardm-stripe) [reply charge.go:558]: Fixed

**Pull request merged**